File: 0429. N-ary Tree Level Order Traversal/Solution0429.py
from collections import deque
from typing import List

class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children

class Solution0429:
    def levelOrder(self, root: Node) -> List[List[int]]:
        res = []
        if not root: return res
        q = deque([root]) 
        while q:
            level = []
            len_ = len(q)
            for i in range(len_):
                cur_node = q.popleft()
                level.append(cur_node.val)
                for c in cur_node.children:
                    q.append(c)
            res.append(level)
        return res

File: 0429. N-ary Tree Level Order Traversal/Solution0429.cpp
#include<vector>
#include<iostream>
#include<queue>

using namespace std;

class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};


class Solution0429 {
public:
    vector<vector<int>> levelOrder(Node* root) {
        vector<vector<int>> res;
        if (root == nullptr) return res;
        queue<Node*> queue;
        queue.push(root);
        while (!queue.empty()) {
            int size = queue.size();
            vector<int> curLayer;
            for (int i = 0; i < size; i++) {
                Node* tmp = queue.front(); queue.pop();
                curLayer.push_back(tmp->val);
                for (Node* child : tmp->children) {
                    queue.push(child);
                }
            }
            res.push_back(curLayer);
        }
        return res;
    }
};